repos/system_upgrade/el7toel8/actors/localreposinhibit/actor.py

```diff
@@ -0,0 +1,64 @@
+from leapp import reporting
+from leapp.actors import Actor
+from leapp.models import TMPTargetRepositoriesFacts, UsedTargetRepositories
+from leapp.reporting import Report
+from leapp.tags import IPUWorkflowTag, TargetTransactionChecksPhaseTag
+
+
+class LocalReposInhibit(Actor):
+    """Inhibits the upgrade if local repositories were found."""
+
+    name = "local_repos_inhibit"
+    consumes = (
+        UsedTargetRepositories,
+        TMPTargetRepositoriesFacts,
+    )
+    produces = (Report,)
+    tags = (IPUWorkflowTag, TargetTransactionChecksPhaseTag)
+
+    def process(self):
+        # fmt: off
+        used_target_repos = next(self.consume(UsedTargetRepositories)).repos
+        target_repos = next(self.consume(TMPTargetRepositoriesFacts)).repositories
+        # fmt: on
+        target_repo_id_to_url_map = {
+            repo.repoid: repo.baseurl
+            for repofile in target_repos
+            for repo in repofile.data
+            if repo.baseurl
+        }
+        if any(
+            target_repo_id_to_url_map[repo.repoid].startswith("file:")
```

Comment:
I assuming that it is always the repo.repoid exists inside target_repo_id_to_url_map. If there could cases when this is not true, I have to redesign a code a bit @pirat89

Reply:
@ZhukovGreen no. In your case you could hit error because not every repository have baseurl.